### PR TITLE
Easier detectable errors for non-needle failures

### PR DESCRIPTION
**Pull request by coolo**
Both wait_serial and die '' were pretty hard to detect on the webui
as they only reveal themselves to the knowing as red screens with a
hint in the log file. That's very hard for beginners to grasp

So change that: In case of wait_serial and die we record a 'fail' text result
together with a screenshot.

See https://progress.opensuse.org/issues/6526

taking this test

```
sub run {
   wait_serial 'hallo', 3;
   die "No Hallo";
}

```

openQA result in master:
before

openQA result after my PR:
before1

### CODE REVIEW

File: basetest.pm [hunk 1]
```diff
@@ -94,16 +94,13 @@ sub record_screenmatch {
...
-
     my $h          = $self->_serialize_match($match);
     my $properties = $match->{needle}->{properties} || [];
     my $result     = {
         needle     => $h->{name},
         area       => $h->{area},
-        tags       => [@$tags],                                  # make a copy
-        screenshot => sprintf("%s-%d.png", $testname, $count),
+        tags       => [@$tags],                               # make a copy
+        screenshot => $self->$self->next_resultname('png'),
```
Comment (okurz) [inline basetest.pm]: Is this related? Why?

File: basetest.pm [hunk 2]
```diff
@@ -164,11 +161,6 @@ sub _serialize_match {
             $na->{$i} = $a->{$i};
         }
         $na->{similarity} = int($a->{similarity} * 100);
-        if ($a->{diff}) {
-            my $imgname = sprintf("%s-%d-%s-diff%d.png", $testname, $count, $name, $diffcount++);
-            $a->{diff}->write(join('/', bmwqemu::result_dir(), $imgname));
-            $na->{diff} = $imgname;
-        }
```
Comment (okurz) [inline basetest.pm]: why delete?

File: basetest.pm [hunk 3]
```diff
@@ -291,13 +280,29 @@ sub runtest {
         $self->post_run_hook();
     };
     if ($@ || ($self->{result} || '') eq 'fail') {
-        my $msg = "test $name " . ($@ ? 'died: ' . $@ : 'failed');
-        warn $msg;
```
Comment (okurz) [inline basetest.pm]: Can you also give an example who the log output changes?

**Comment by okurz** [comment]
-1: really nice, minor issues. Probably doesn't need code change to resolve my questions

File: basetest.pm (hunk 1 above)
Comment (coolo) [inline basetest.pm]: good thing you spotted the $self->$self :)

File: basetest.pm (hunk 2 above)
Comment (coolo) [inline basetest.pm]: it's a very old idea - and these diffs aren't even uploaded to openQA.

File: basetest.pm (hunk 3 above)
Comment (coolo) [inline basetest.pm]: who?

Comment (okurz) [inline basetest.pm]: s/who/how

Comment (coolo) [inline basetest.pm]: ah, it doesn't change at all. I just moved the trigraph into the if

File: basetest.pm (hunk 2 above)
Comment (coolo) [inline basetest.pm]: actually the diff elements are gone since 99ff3f5d9b572e779b7ddbeb303cb29862925d39 - Jul 2014

File: basetest.pm [hunk 4]
```diff
@@ -291,13 +280,29 @@ sub runtest {
...
+                $self->_result_add_screenshot($result);
+            }
+            my $details = {result => 'fail'};
+
+            my $text_fn = $self->next_resultname('txt');
+            open my $fd, ">", bmwqemu::result_dir() . "/$text_fn";
+            print $fd "# Test died:\n$@\n";
+            close $fd;
+            $details->{text}  = $text_fn;
+            $details->{title} = 'DIE';
```
Comment (aaannz) [inline basetest.pm]: Looking at the screen shot, I'm not sure we want openQA to scream 'DIE' on unsuspecting user. Too bad `$text_fn` won't fit, but what about putting there first few chars that fit and `...` to signalize hint text?

Comment (sysrich) [inline basetest.pm]: Discussed with @coolo , my vote is for replacing 'DIE' with a red "Failed" thumbnail

Comment (coolo) [inline basetest.pm]: a4ec8a2
